bundle/deploy/check_running_resources.go

```diff
@@ -35,82 +34,55 @@ func (l *checkRunningResources) Apply(ctx context.Context, b *bundle.Bundle) dia
 	if !b.Config.Bundle.Deployment.FailOnActiveRuns {
 		return nil
 	}
-
-	tf := b.Terraform
-	if tf == nil {
-		return diag.Errorf("terraform not initialized")
-	}
-
-	err := tf.Init(ctx, tfexec.Upgrade(true))
-	if err != nil {
-		return diag.Errorf("terraform init: %v", err)
-	}
-
-	state, err := b.Terraform.Show(ctx)
+	w := b.WorkspaceClient()
+	err := checkAnyResourceRunning(ctx, w, &b.Config.Resources)
```

Comment:
We need all resources which are defined before the deployment because we might have removed something and it's not in Config.Resources anymore and we want to check that these resources are not in running state still

Reply:
Such resources will be added to the config by the `terraform.Load` (with IDs and ModifiedStatus = deleted). That's another reason I've added `terraform.Load` before this mutator

Reply:
Based on the offline discussion also added logic that prevents saving resources that were locally deleted - in case we ever to `terraform.Load` before `terraform.Write` (currently we don't do that though)